src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/CallChainProfile.cs

```diff
@@ -0,0 +1,331 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+using ILCompiler.IBC;
+
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+namespace ILCompiler
+{
+    // This is a sample of the Json format the call chain data is stored in:
+    //
+    //{
+    //    "Microsoft.CodeAnalysis.CSharp.BinderFactory!GetBinder": [
+    //        [
+    //            "Microsoft.CodeAnalysis.CSharp.BinderFactory!GetBinder",
+    //            "Microsoft.CodeAnalysis.CSharp.BinderFactory+BinderFactoryVisitor!VisitCompilationUnit"
+    //        ],
+    //        [
+    //            1,
+    //            2
+    //        ]
+    //    ],
+    //    "Microsoft.CodeAnalysis.CSharp.BinderFactory+BinderFactoryVisitor!VisitCompilationUnit": [
+    //        [
+    //            "System.Lazy`1[System.__Canon]!CreateValue"
+    //        ],
+    //        [
+    //            1
+    //        ]
+    //    ],
+    //}
+    public class CallChainProfile
+    {
+        IEnumerable<ModuleDesc> _referenceableModules;
```

Comment:
Nit - this could be probably readonly. I'm also not sure what's the current coding style policy but the last time I looked we strived to always explicitly specify access right so I'd use a 'private' modifier. The same for _resolvedProfileData; _resolveFails could probably also be readonly.

Reply:
I made these `private readonly`